tests/unit/test_server_timing.py

```diff
@@ -0,0 +1,97 @@
+import pytest
+
+from globus_cli.termio.server_timing import (
+    Draft2017Parser,
+    Metric,
+    ServerTimingParseError,
+)
+
+
+@pytest.mark.parametrize(
+    "metricstr, expect",
+    (
+        ("a=1", Metric(name="a", duration=1.0)),
+        (
+            'a=1; "complex desc"',
+            Metric(name="a", duration=1.0, description="complex desc"),
+        ),
+        (
+            "some-str",
+            Metric(name="some-str"),
+        ),
+        (
+            'some-str; "with a description"',
+            Metric(name="some-str", description="with a description"),
+        ),
+        (
+            'some-str=52.4; "with a description"',
+            Metric(name="some-str", description="with a description", duration=52.4),
+        ),
+    ),
+)
+def test_draft2017_parse_of_single_metric(metricstr, expect):
+    parser = Draft2017Parser()
+    assert parser.parse_single_metric(metricstr) == expect
+
+
+@pytest.mark.parametrize("metricstr", ("a; b; c", "", "a=foo"))
+def test_draft2017_parse_single_metric_errors(metricstr):
+    parser = Draft2017Parser()
+    with pytest.raises(ServerTimingParseError):
+        parser.parse_single_metric(metricstr)
+
+
+@pytest.mark.parametrize(
+    "metricstr, expect",
+    (
+        (
+            "a=1, b=2.2",
+            [
+                Metric(name="a", duration=1.0),
+                Metric(name="b", duration=2.2),
+            ],
+        ),
+        (
+            'a=1, b, c=2.2; "callout"',
```

Comment:
I haven't seen a case that exercises the `.strip()` calls. Perhaps:

```python
' a=1 , b , c=2.2 ; "callout" '
```

(It might be worthwhile to test spaces on either side of the equals sign...don't know if the spec allows for that, though.)

Reply:
I decided to make the test slightly more stringent and put in ` a = 1 `, which then promptly failed to parse as I wanted ("a " vs "a").
So the test change also includes _another_ call to `strip()` to get that one to come out as `"a"`.

That said, overall I feel good about where the tests put us on this front.